### ISSUE TITLE
Improve build instructions about `porto` and `make goporto`

### ISSUE THREAD

**Issue by bertysentry** [open-telemetry/opentelemetry-collector-contrib#6498]
## Issue

The build may fail with the below error message:

```
Generated code is out of date, please run "make generate" and commit the changes in this PR.
Error: Process completed with exit code 1.
```

The message is wrong, it should recommend running `make goporto`.

## How to reproduce

Commit a new Go source file like below:

```go
package groupbyattrsprocessor // Nothing here
```

and run `make ci`. You should get the wrong error message.

Then try running `make goporto`, which will fail with an error message complaining about not finding *porto*.

## Expected behavior

Error message should recommend running `make goporto`. 

Also, the build instructions should mention that the contributor may need to run `make goporto`, and this operation requires the proper installation of [porto](https://github.com/jcchavezs/porto):

```bash
go install github.com/jcchavezs/porto/cmd/porto@latest
```

## Version

0.40.0

## Environment

Build on Linux, with Go 1.17.2

**Comment by mx-psi** [comment]
@jcchavezs would you want to take this one?

**Comment by bertysentry** [comment]
Actually, if there was a `make prepare` command that would run all checks necessary before submitting a PR, that would be awesome.

**Comment by jpkrohling** [comment]
I understand this issue here is only about the documentation/error message. A new feature would deserve a new issue ;-)

**Comment by bertysentry** [comment]
@jpkrohling Is the installation of **goporto** still a prerequisite for building the project? If so, this issue still needs to be fixed. I have very little knowledge on the build script of the project and I don't feel I'll be able to fix the `Makefile` with a proper error message...

**Comment by mx-psi** [comment]
AFAICT this is still a problem, removing the Stale label

**Comment by bertysentry** [comment]
Actually, it looks like the error message has been updated:

In **.github\workflows\build-and-test.yml**:

```yaml
      - name: Porto
        run: |
          make -j2 goporto
          git diff --exit-code || (echo 'Porto links are out of date, please run "make goporto" and commit the changes in this PR.' && exit 1)
```

So we would simply need to update some documentation on the technical prerequisites to build the project. I couldn't find any information on building the project anywhere. Is there such thing?

**Comment by bertysentry** [comment]
So, the main issue was apparently fixed in PR #8099 by @bogdandrutu

**Comment by mx-psi** [comment]
We have [this page](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#required-tools) and [porto is installed by `make install-tools`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/261a3e7e438fe6c8e70cfb0b02641f92649ef555/Makefile#L236) so maybe there is nothing to do after all :) 

I will leave it up to you @bertysentry whether to close this.

**Comment by bertysentry** [comment]
You're right, @mx-psi! **porto** is now installed with `make install-tools`. This was fixed by @codeboten in PR #7902, so we're good! 👌
